simulation/simtypes/app.go

```diff
@@ -24,10 +25,3 @@ type AccountKeeper interface {
 	GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI
 	// GetAllAccounts(ctx sdk.Context) []authtypes.AccountI
 }
-
-type BankKeeper interface {
-	GetBalance(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin
-	GetAllBalances(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins
-	// TODO: Revisit
-	SpendableCoins(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins
-}
```

Comment:
Why is it not possible to define the desired methods on this interface?

Allowing them all by default doesn't seem preferable to me. We should be limiting the interface only to the functionality that we need for the purposes of reducing complexity and the risk of accidentally misusing undesired functionality.

Reply:
hmm, so it is no need to handle this issue, right?

Reply:
@p0mvn so we can define the methods on the interface but there are restrictions to it. For instance if you try to add his method `SetDenomMetaData(ctx sdk.Context, denomMetaData banktypes.Metadata)`, the rest of the simulator breaks because if you check x/incentives/simulation/operations.go we use `bk stakingTypes.BankKeeper`  to get bankkeeper. 

this PR i think removes `stakingTypes.BankKeeper` and uses a unified bankKeeper interface, we donot necessarily have to use bankKeeper from cosmos-sdk instead we can keep the interface and use simtypes.BankKeeper